test/clients.go

```diff
@@ -173,9 +173,11 @@ func CleanupAll(contexts ...*Context) {
 	}
 }
 
-// Cleanup iterates through the list of registered CleanupFunc functions and calls them
+// Cleanup iterates through the list of registered CleanupFunc functions in reverse order and calls them
 func (ctx *Context) Cleanup() {
-	for _, f := range ctx.CleanupList {
+	last := len(ctx.CleanupList)-1
```

Comment:
The AddToCleanup function adds them to this list in reverse order so you're effectively changing it to be removing NOT in reverse order.

Reply:
I didn't notice that. I'll restore it.

Reply:
Restored.